afl-cmin.py

```diff
@@ -0,0 +1,687 @@
+        sys.exit(1)
+
+    trace_dir = os.path.join(args.output, '.traces')
+    shutil.rmtree(trace_dir, ignore_errors=True)
+    try:
+        os.rmdir(args.output)
+    except OSError:
+        pass
+    if os.path.exists(args.output):
+        logger.error(
+            'directory "%s" exists and is not empty - delete it first',
+            args.output)
+        sys.exit(1)
+    if args.crash_dir and not os.path.exists(args.crash_dir):
+        os.makedirs(args.crash_dir)
+    os.makedirs(trace_dir)
+
+    logger.info('use %d workers (-T)', args.workers)
+
+
+def detect_type_code(size):
+    for type_code in ['B', 'H', 'I', 'L', 'Q']:
+        if 256**array.array(type_code).itemsize > size:
+            return type_code
+
+
+def afl_showmap(input_path=None, batch=None, afl_map_size=None, first=False):
+    assert input_path or batch
+    # yapf: disable
+    cmd = [
+        afl_showmap_bin,
+        '-m', str(args.memory_limit),
+        '-t', str(args.time_limit),
+        '-Z', # cmin mode
+    ]
+    # yapf: enable
+    found_atat = False
+    for arg in args.args:
+        if '@@' in arg:
+            found_atat = True
+
+    if args.stdin_file:
+        assert args.workers == 1
+        input_from_file = True
+        stdin_file = args.stdin_file
+        cmd += ['-H', stdin_file]
+    elif found_atat:
+        input_from_file = True
+        stdin_file = os.path.join(args.output, f'.input.{os.getpid()}')
+        cmd += ['-H', stdin_file]
+    else:
+        input_from_file = False
+
+    if batch:
+        input_from_file = True
+        filelist = os.path.join(args.output, f'.filelist.{os.getpid()}')
+        with open(filelist, 'w') as f:
+            for _, path in batch:
+                f.write(path + '\n')
+        cmd += ['-I', filelist]
+        output_path = os.path.join(args.output, f'.showmap.{os.getpid()}')
+        cmd += ['-o', output_path]
+    else:
+        if input_from_file:
+            shutil.copy(input_path, stdin_file)
+        cmd += ['-o', '-']
+
+    if args.frida_mode:
+        cmd += ['-O']
+    if args.qemu_mode:
+        cmd += ['-Q']
+    if args.unicorn_mode:
+        cmd += ['-U']
+    if args.nyx_mode:
+        cmd += ['-X']
+    if args.edge_mode:
+        cmd += ['-e']
+    cmd += ['--', args.exe] + args.args
+
+    env = os.environ.copy()
+    env['AFL_QUIET'] = '1'
+    env['ASAN_OPTIONS'] = 'detect_leaks=0'
+    if first:
+        logger.debug('run command line: %s', subprocess.list2cmdline(cmd))
+        env['AFL_CMIN_ALLOW_ANY'] = '1'
+    if afl_map_size:
+        env['AFL_MAP_SIZE'] = str(afl_map_size)
+    if args.crash_only:
+        env['AFL_CMIN_CRASHES_ONLY'] = '1'
+    if args.allow_any:
+        env['AFL_CMIN_ALLOW_ANY'] = '1'
+
+    if input_from_file:
+        p = subprocess.Popen(cmd,
+                             stdout=subprocess.PIPE,
+                             env=env,
+                             bufsize=1048576)
```

Comment:
Maybe the value should be a global, especially since you use it twice

Reply:
I didn't do so because the value is not so important IMHO.

Reply:
I mean you use it twice, it cries for a variable. That way you can also make clear, why you set it to that number